aspnet/mvc/models/validation/sample/MoviesController.cs

```diff
@@ -0,0 +1,62 @@
+using System.Net.Http;
+using System.Linq;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.Data.Entity;
+using MVCMovie.Models;
+using System.Net;
+using System.Web.Http;
+using System;
+
+namespace MVCMovie.Controllers
+{
+public class MoviesController : Controller
+{
+    private MVCMovieContext _context;
+    public MoviesController(MVCMovieContext context)
+    {
+        _context = context;
+    }
+
+    [HttpPost]
+    [ValidateAntiForgeryToken]
+    public IActionResult Edit(
+        string title, DateTime releaseDate, 
+        string description, bool preorder)
+    {
+        DateTime modifiedReleaseDate = releaseDate;
+        if (releaseDate == null) {
+            modifiedReleaseDate = DateTime.Today;
+        }
+
+        var movie = new Movie
+        {
+            Title = title,
+            ReleaseDate = modifiedReleaseDate,
+            Description = description,
+            Preorder = true
+        };
+        
+        if (movie.Preorder) {
+            if (modifiedReleaseDate.Date <= DateTime.Today)
+            {
+                ModelState.AddModelError("preorder", 
+                    "Movies already released may not also be preorder movies.");
+            }                
+        }
+                        
+        TryValidateModel(movie);
+
+        if (ModelState.IsValid)
+        {
+            _context.Movie.Add(movie);
+            _context.SaveChanges();           
+            return RedirectToAction("Index","Home");    
+        }
+        else
+        {
+            return HttpBadRequest(ModelState);
```

Comment:
This code doesn't fit in a non-API action. I mentioned the possibility of actually returning model errors directly in the context of the document text.

Expected code here is `View(model)`. The `Edit.cshtml` file should contain HTML or tag helpers to list the errors but refer to the Forms document rather than go into depth about `@Html.ValidationSummary()` and so on.